ASM/README.md

```diff
@@ -1,11 +1,55 @@
+## C: n64 toolchain
+### Prerequisites
+Recompiling the C code for randomizer requires the N64 development tools to be installed: <https://github.com/glankk/n64>. There are several ways to do this depending on your platform.
+- **Windows**:
+  - **Without WSL**: [Download this zip archive](https://discord.com/channels/274180765816848384/442752384834469908/1085678948614144081) and extract the `n64` folder into the `tools` directory alongside armips.
+    - Download and install [MSYS2](https://www.msys2.org/#installation).
+      1. Accept the defaults in the installer.
+      2. After the installer completes a terminal window will open.
+      3. In the terminal type `pacman -Syy make` and press Enter.
+      4. Make sure it lists `make` for installation and press Enter again to confirm.
+      5. After the installation finishes you can close the terminal window.
+      6. In the Start search bar type "Environment Variables" and click "Edit the system environment variables".
+      7. Near the bottom click on the button labeled "Environment Variables...".
+      8. In the new window in the top section look for the variable called "Path" and click it.
+      9. Click the "Edit..." button below the box you selected "Path" in.
+      10. Click on "New" on the right side of the new window.
+      11. Type `C:\msys64\usr\bin` and press Enter.
+      12. Click "OK" on all three windows.
+      13. You will now be able to compile the randomizer's C code from CMD, PowerShell, and MSYS2's terminal.
+  - **Using WSL**: Install the latest Debian Linux from the Windows Store and follow the below instructions for Debian.
+- **Debian**: [Follow this how-to](https://practicerom.com/public/packages/debian/howto.txt) on adding the toolchain's package repository and installing the pre-built binaries.
+  - You will also need to run `apt install build-essential` or `apt install make` if `make` is not installed.
+- **Any platform with a gcc compiler**: Build from source from the [glankk/n64](https://github.com/glankk/n64) repository. Simply follow the readme.
+  - The dependency install script may not install all the necessary libraries depending on your OS version. Take a look at the output from the configure step to see if anything is missing.
+  - It is easiest if you use `--prefix=/the/path/to/OoT-Randomizer/ASM/tools` for the `./configure` step. This will install all the toolchain in a way the build script can use, however this is inconvenient if you plan to use the toolchain for other projects as well.
+  - If you are trying to update the toolchain this way, it is easiest to just delete your local copy of the repository and clone it again to ensure all the packages get updated and are compatible.
+
+
+You can substitute using the `tools` folder with adding the `n64/bin` folder to your environment PATH if you need an advanced setup.
+### Running
+To recompile the C modules, use `python build.py --compile-c`.
```

Comment:
I would either specify to cd into the ASM folder, or change the command to `python ASM/build.py` (the latter of which is what I typically do)

Reply:
I suppose I could do this, but I kind of assumed everything to be done from the same directory as the README itself.

Reply:
I'd probably prefer saying something like "in this directory" or "the same directory as this README" due to the above. It's just too awkward to have only this one command reference the directory when nothing else really does.

Reply:
Yeah, I'm good with saying either of those. I guess to me the default is running commands from the repo root

Reply:
To be fair, that's usually where the README.md and/or CONTRIBUTING.md are.